packages/app-builder/src/routes/_builder+/settings+/analytics+/filters.tsx

```diff
@@ -0,0 +1,293 @@
+import { useLoaderData } from '@remix-run/react';
+import { createColumnHelper, getCoreRowModel } from '@tanstack/react-table';
+import { type Namespace } from 'i18next';
+import { useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ButtonV2, Modal, Table, useTable } from 'ui-design-system';
+import { Icon } from 'ui-icons';
+
+export const handle = {
+  i18n: ['common', 'settings'] satisfies Namespace,
+};
+type FilterRow = {
+  id: string;
+  tableId: string;
+  associatedObject: string;
+  definition: string;
+  kind: 'trigger' | 'ingested';
+  field?: string;
+  path?: string[];
+  name?: string;
+};
+
+type TriggerFieldItem = {
+  tableId: string;
+  tableName: string;
+  fieldName: string;
+  label: string;
+};
+
+type LinkPivotFieldItem = {
+  baseTableId: string;
+  pathLinks: string[];
+  fieldName: string;
+  label: string;
+};
+
+export async function loader({ request }: LoaderFunctionArgs) {
+  const { authService } = initServerServices(request);
+  const { user, dataModelRepository } = await authService.isAuthenticated(request, {
+    failureRedirect: getRoute('/sign-in'),
+  });
+
+  if (!isAdmin(user)) {
+    return redirect(getRoute('/'));
+  }
+  const dataModel = await dataModelRepository.getDataModel();
+
+  const exportedEntries = await Promise.all(
+    dataModel.map(async (table) => {
+      const exported = await dataModelRepository.getDataModelTableExportedFields(table.id);
+      return [table.id, exported] as const;
+    }),
+  );
+  const exportedFieldsByTable = Object.fromEntries(exportedEntries);
+
+  const tableIdToName = new Map<string, string>(dataModel.map((t) => [t.id, t.name] as const));
+
+  const filters: FilterRow[] = Object.entries(exportedFieldsByTable).flatMap(
+    ([tableId, exported]) => {
+      const tableName = tableIdToName.get(tableId) ?? '';
+
+      const triggerFilters: FilterRow[] = (exported.triggerObjectFields ?? []).map((field) => ({
+        id: `${tableId}::trigger::${field}`,
+        tableId,
+        associatedObject: tableName,
+        definition: tableName ? `${tableName}.${field}` : field,
+        kind: 'trigger' as const,
+        field,
+      }));
+
+      const ingestedFilters: FilterRow[] = (exported.ingestedDataFields ?? [])
+        .filter((field): field is NonNullable<typeof field> => Boolean(field?.name))
+        .map((field) => {
+          const pathArr = Array.isArray(field.path) ? field.path : [];
+          const pathStr = pathArr.join('->');
+          return {
+            id: `${tableId}::ingested::${pathStr}.${field.name}`,
+            tableId,
+            associatedObject: tableName,
+            definition: `${tableName}->${pathStr}.${field.name}`,
+            kind: 'ingested' as const,
+            field: field.name,
+            name: field.name,
+            path: pathArr,
+          };
+        });
+
+      return [...triggerFilters, ...ingestedFilters];
+    },
+  );
```

Comment:
Can it not be exported as an external function for better readability and reducing loader code size?